addons/hr_contract/models/hr_employee.py

```diff
@@ -146,12 +146,18 @@ def _get_calendar_attendances(self, date_from, date_to):
         if not valid_contracts:
             return super()._get_calendar_attendances(date_from, date_to)
         employee_timezone = timezone(self.tz) if self.tz else None
-        return valid_contracts.resource_calendar_id\
-            .with_context(employee_timezone=employee_timezone)\
-            .get_work_duration_data(
-                date_from,
-                date_to,
-                domain=[('company_id', 'in', [False, self.company_id.id])])
+        duration_data = {'days': 0, 'hours': 0}
+        for contract in valid_contracts:
+            calendar = contract.resource_calendar_id or contract.company_id.resource_calendar_id
+            contract_duration_data = calendar\
+                .with_context(employee_timezone=employee_timezone)\
+                .get_work_duration_data(
+                    max(date_from, contract.date_start),
+                    min(date_to, contract.date_end),
```

Comment:
I think this will give a traceback since `date_from`/`date_to` are `datetime` objects while the contract dates are `date` objects.
`contract.date_end` might be `False` for long-terms contracts so it should reflect that as well :) 
```suggestion
                    max(date_from, datetime.combine(contract.date_start, time.min)),
                    min(date_to, datetime.combine(contract.date_end, time.max)) if contract.date_end else date_to,
```

Reply:
Good idea, though I feel like also using a `or date.max` instead of the `_ if _ else _` would be cleaner, if only to keep the diff below 100 chars a line
```python
    min(date_to, datetime.combine(contract.date_end or date.max, time.max)),
```

Reply:
Works as well so fine by me! :D With those changes everything LGTM